utils/selectionStateUtils.ts

```diff
@@ -45,13 +45,56 @@ const updateSelectionState = (
   return updatedState;
 };
 
-const selectionStateToString = (selectionState: SelectionState): string => {
+const selectionStateTrueToString = (
+  selectionState: SelectionState | undefined,
+): string => {
+  if (selectionState === undefined) return '';
   const stringRepresentation = Object.entries(selectionState)
     .filter(([, value]) => value)
     .map(([key]) => key)
     .join(', ');
 
-  return stringRepresentation.length > 0 ? `${stringRepresentation},` : '';
+  return stringRepresentation.length > 0 ? `${stringRepresentation}, ` : '';
+};
+
+const stateToValidatedState = (
+  state1: SelectionState | undefined,
+  state2: SelectionState | undefined,
```

Comment:
```suggestion
  state1: SelectionState | undefined,
  state2: SelectionState | undefined,
```

trouver des noms de variables qui donnent un sens à ce que ça fait.